src/library/stateful-computation.lisp.orig

```diff
@@ -0,0 +1,88 @@
+(in-package #:coalton-library)
+
+(coalton-toplevel
+  ;;
+  ;; StatefulComputation
+  ;;
+
+  (define-type (StatefulComputation :state :value)
+    "A computation of a value which may affect the state.
+Represented as a closure from initial state to updated state and value."
+    (StatefulComputation (:state -> (Tuple :state :value))))
+
+  (declare stateful-computation-put (:state -> (StatefulComputation :state Unit)))
+  (define (stateful-computation-put state)
+    "A StatefulComputation with state set to be given state. The returned value is Unit."
+    (StatefulComputation (fn (_) (Tuple state Unit))))
+
+  (declare stateful-computation-get (StatefulComputation :state :state))
+  (define stateful-computation-get
+    "A StatefulComputation which returns the current state as the value."
+    (StatefulComputation (fn (state) (Tuple state state))))
+
+<<<<<<< HEAD
```

Comment:
uh oh

Reply:
embarrassing. this file is leftover from a merge and I didn't see it in the diff.